src/base/bittorrent/peeraddress.cpp

```diff
@@ -68,3 +68,23 @@ QString PeerAddress::toString() const
         : ip.toString();
     return (ipStr + ':' + QString::number(port));
 }
+
+bool BitTorrent::operator==(const BitTorrent::PeerAddress &left, const BitTorrent::PeerAddress &right)
+{
+    return (left.ip == right.ip) && (left.port == right.port);
+}
+
+uint BitTorrent::qHash(const BitTorrent::PeerAddress &addr, const uint seed)
+{
+    return (::qHash(addr.ip, seed) ^ ::qHash(addr.port, seed));
+}
+
+bool BitTorrent::operator==(const BitTorrent::PeerEndpoint &left, const BitTorrent::PeerEndpoint &right)
+{
+    return (left.address == right.address) && (left.connectionType == right.connectionType);
+}
+
+uint BitTorrent::qHash(const BitTorrent::PeerEndpoint &peerEndpoint, const uint seed)
```

Comment:
Functions related to PeerEndpoint should be moved to peerlistwidget.cpp as well.